src/app/core/services/authenticate.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { ApiService, AuthToken, UserDTO } from '../../../backend-client';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticateService {
+  private currentUserSubject: BehaviorSubject<Observable<UserDTO> | null>;
+  public currentUser: Observable<Observable<UserDTO> | null>;
+
+  constructor(private api: ApiService) {
+    this.currentUserSubject = new BehaviorSubject<Observable<UserDTO> | null>(
```

Comment:
ตอนนี้ authController มันยิงไปจริงๆแล้วล้มเหลวอะ เราต้องทำ [api interceptor](https://medium.com/@ryanchenkie_40935/angular-authentication-using-the-http-client-and-http-interceptors-2f9d1540eb8) เพิ่มด้วยมันจะได้ inject token ไปทุกครั้งที่ยิง request
![image](https://user-images.githubusercontent.com/12471844/86486969-1217b680-bd87-11ea-881e-42c17a050f1b.png)